modules/effects/src/effect_creator.ts

```diff
@@ -61,7 +61,9 @@ export function createEffect<
   return effect;
 }
 
-export function getCreateEffectMetadata<T>(instance: T): EffectMetadata<T>[] {
+export function getCreateEffectMetadata<
+  T extends { [props in keyof T]: Object }
```

Comment:
May be `T extends Record<keyof T, Object>` would be better?

Reply:
Yes, it's the same thing. Don't think we used `Record` at the codebase yet.

Reply:
Then fine :)